app/controllers/users/piv_cac_authentication_setup_controller.rb

```diff
@@ -75,7 +77,10 @@ def next_step
     def process_invalid_submission
       @presenter = PivCacAuthenticationSetupErrorPresenter.new(user_piv_cac_form)
       clear_piv_cac_information
-      render :error
+      flash[:error] = @presenter.title
+      flash[:description] = @presenter.description
+      flash[:heading] = @presenter.heading
```

Comment:
Is there a reason we can't use `@presenter.description` and `@presenter.heading` directly in the view?

Reply:
We redirect before showing the view, so we don't have a presenter with the information.

Reply:
I see. I would have suggested passing in the error as a params, and then defining the presenter before rendering the error view:
```ruby
elsif flash[:error].present?
  @presenter = PivCacAuthenticationSetupErrorPresenter.new(params[:error])
  render :error
end
```
The problem is that the presenter class expects to be initialized with an entire form object, even though the only thing it needs from it is its `error_type`. The solution is to change the presenter to be initialized with the error, but that's a refactoring for later. The controller would look like this:
```ruby
def process_invalid_submission
  form_error = user_piv_cac_form.error_type
  @presenter = PivCacAuthenticationSetupErrorPresenter.new(form_error)
  clear_piv_cac_information
  flash[:error] = @presenter.title
  redirect_to setup_piv_cac_url(error: form_error)
end
```

Reply:
Good point. I was trying to keep the error condition out of the URL, but I could have just put that in the flash and rebuilt the presenter from there after the redirect.